packages/tools/examples/dynamicCINETool/index.ts

```diff
@@ -0,0 +1,479 @@
+  container.style.transition = 'opacity 0.3s';
+
+  titleEl.innerHTML = 'Stage 1: Load';
+  titleEl.style.fontWeight = 'bold';
+
+  dropdownLabel.innerHTML = 'Time points to load: ';
+
+  for (let i = 1; i <= MAX_NUM_TIMEPOINTS; i++) {
+    dropDownOptions.push(i);
+  }
+
+  addDropdownToToolbar({
+    id: 'numTimePointsDropdown',
+    options: {
+      values: dropDownOptions,
+      defaultValue: numTimePoints,
+    },
+    container: toolBarEl,
+    onSelectedValueChange: (value) => {
+      numTimePoints = <number>value;
+    },
+  });
+
+  addButtonToToolbar({
+    id: 'btnLoadTimePoints',
+    title: 'Load',
+    container: toolBarEl,
+    onClick: () => {
+      const dropdown = <HTMLSelectElement>(
+        document.getElementById('numTimePointsDropdown')
+      );
+      const btnLoadTimePoints = <HTMLButtonElement>(
+        document.getElementById('btnLoadTimePoints')
+      );
+      const secondStageContainer = <HTMLDivElement>(
+        document.getElementById('secondStageContainer')
+      );
+
+      container.style.opacity = '0.4';
+      dropdown.disabled = true;
+      btnLoadTimePoints.disabled = true;
+      secondStageContainer.style.opacity = '1';
+
+      secondStageContainer.addEventListener(
+        'transitionend',
+        () => loadTimePoints(numTimePoints),
+        { once: true }
+      );
+    },
+  });
+
+  return container;
+}
+
+function createSecondStageLayout() {
+  const container = document.createElement('div');
+  const titleEl = createChildEl(container, 'div');
+  const firstRowEl = createChildEl(container, 'div');
+  const secondRowEl = createChildEl(container, 'div');
+  const infoEl = createChildEl(secondRowEl, 'span');
+
+  container.id = 'secondStageContainer';
+  container.style.opacity = '0';
+  container.style.transition = 'opacity 0.3s';
+
+  titleEl.innerText = 'Stage 2: Interact';
+  titleEl.style.fontWeight = 'bold';
+
+  infoEl.innerText = 'Global 4D Cine ';
+
+  addSliderToToolbar({
+    id: 'fpsSlider',
+    title: ` Time points per second: ${framesPerSecond}`,
+    range: [1, 100],
+    defaultValue: framesPerSecond,
+    container: firstRowEl,
+    onSelectedValueChange: (value) => {
+      framesPerSecond = Number(value);
+      startCine();
+    },
+    updateLabelOnChange: (value, label) => {
+      label.innerText = ` Time points per second: ${value}`;
+    },
+  });
+
+  addButtonToToolbar({
+    title: 'Play Clip',
+    container: secondRowEl,
+    onClick: () => {
+      startCine();
+    },
+  });
+
+  addButtonToToolbar({
+    title: 'Stop Clip',
+    container: secondRowEl,
+    onClick: () => {
+      csToolsUtilities.cine.stopClip(activeElement);
+    },
+  });
+
+  return container;
+}
+
+function initLayout() {
+  const content = document.getElementById('content');
+  const stagesContainer = document.createElement('div');
+  const firstStageContainer = createFirstStageLayout();
+  const secondStageContainer = createSecondStageLayout();
```

Comment:
this file has 70% UI setup, can we refactor the createFirst/Seconf functions outside maybe?

Reply:
Updated